validator/project_validator.go

```diff
@@ -1898,3 +1899,21 @@ func parseS3PullParameters(c model.PluginCommandConf) (task, bv string, err erro
 	}
 	return task, bv, nil
 }
+
+// checkTaskTimeout checks if all tasks contain a timeout
+func checkTaskTimeout(project *model.Project) ValidationErrors {
```

Comment:
Some unit tests are gonna need to be updated to expect this warning now - a couple are failing.  Once that's done add that to the testing section of the description of this PR as well. Looks good though!